src/register.php

```diff
@@ -27,40 +27,27 @@
 				exit("This email is already registered");
 			}
 
-			// collecting information for validation
-			$time = time(); // in unix format
-			$validation = md5($email . $time . rand());
-			// ignore IP for now due to ip2long() not being able to handle IPv6 addresses
-			// TODO: remove this column from the database
-			$IP = 0;
-
-			$sth = $dice->dbconn->prepare("SELECT email FROM pending_validations WHERE email=?");
-			$sth->bind_param('s', $email);
-			$sth->execute() or trigger_error($mysqli->error);
+			// generating a random token for registration validation
+			// the DB table allows entries up to 32 chars
+			$validation = base64_encode(random_bytes(24));
 
-			$isNewRegistration = $sth->num_rows === 0;
-			$sth->close();
-			// insert or update pending validation
-			if ($isNewRegistration) {
-				$sql = "INSERT INTO pending_validations (email, validation_key, time_stamp, IP) VALUES (?, ?, FROM_UNIXTIME(?), ?)";
-				$sth = $dice->dbconn->prepare($sql);
-				$sth->bind_param('ssss', $email, $validation, $time, $IP);
-				$sth->execute();
-			} else {
-				$sql = "UPDATE pending_validations SET validation_key=?, time_stamp=FROM_UNIXTIME(?), IP=? WHERE email=?";
-				$sth = $dice->dbconn->prepare($sql);
-				$sth->bind_param('ssss', $validation, $time, $IP, $email);
-				$sth->execute();
-			}
+			// ignore IP for now due to ip2long() not being able to handle IPv6 addresses
+			// TODO: remove this column from the database, remove time_stamp as well
+			// because it is ignored as well if the timestamp doesn't provide much value
+			$sql = "REPLACE INTO pending_validations (email, validation_key, IP) VALUES (?, ?, 0)";
```

Comment:
From
https://github.com/triplea-game/dice-server/blob/9b4a45e759e6a36bee332b1f64ed36c157ae845c/config/db/0000_initialize.sql#L19
it seems the time_stamp value is auto-filled (and not used as it appears), so no need to worry about that